### ISSUE TITLE
freetype: need to support png support for some emoji font

### ISSUE THREAD

**Issue by mitchellh** [hexops/mach#466]
I'm still using my own freetype zig fork for now and haven't had time to contribute (sorry!) but I ran into this and I think you will too. Some Emoji fonts such as Noto (Google) use embedded pngs which require freetype to be built with png support. 

This is a bit of a can of worms so I'll explain it all here so you can avoid the footguns:

  1. Freetype will require `libpng`
  2. libpng will require `zlib`
  3. Freetype by default (and current Mach config) compiles in its own vendored zlib sources. So doing step 2 will result in duplicate symbols. You'll want to set `FT_CONFIG_OPTION_SYSTEM_ZLIB=1` for freetype and link in zlib that you used for step 2.

To test it all works, you'll need to load a glyph from a png font, Noto works well.

**Comment by slimsag** [comment]
Embedded PNGs in fonts! I had no idea, that's nasty :/ I agree we should fix this in the freetype bindings as you suggest.

In Mach itself, we're hoping to render glyphs ourselves on the GPU using freetype just to extract beziers - I was hoping we could use Noto too but I guess we'll need to find a fallback approach for whichever glyphs only have pre-rendered PNGs. Crazy.

**Comment by mitchellh** [comment]
Yeah Noto not only uses embedded PNGs, but they're fixed at a single resolution. Right now I'm putting them at full resolution into a texture atlas and downsampling in a shader, but I might move to downsampling a bit on the CPU since its a bit wasteful how large they are.

**Comment by slimsag** [comment]
https://github.com/adobe-fonts/noto-emoji-svg looks interesting?

**Comment by mitchellh** [comment]
Maybe! I haven’t used SVG fonts yet, I’ve only read about them in the FT docs. I know that FT doesn’t include any SVG renderer but you can get the SVG data and probably pass that into a shader…
